components/director/internal/open_resource_discovery/validation.go

```diff
@@ -1107,14 +1118,22 @@ func areThereEntryPointDuplicates(entryPoints []gjson.Result) bool {
 
 func isValidDate(date *string) validation.RuleFunc {
 	return func(value interface{}) error {
-		var err error
-		if _, err = time.Parse("2006-01-02T15:04:05Z0700", *date); err == nil {
-			return nil
-		} else if _, err = time.Parse("2006-01-02T15:04:05Z07:00", *date); err == nil {
-			return nil
-		}
-		return errors.New("invalid date")
+		return validateDate(date)
+	}
+}
+
+func validateDate(date *string) error {
```

Comment:
You can make the date argument `interface{}` and then cast it to string or *string.

If it is not string or *string you will return the relevant error.

This way you will not need to use crossFieldRule in validateORDChangeLogEntries to workaround type issues. 

Also you will not need to invoke the method manually with the value in the inlined rules like:

```go
validation.By(isValidDate(api.SunsetDate))
```

Will be simplified to:
```go
validation.By(isValidDate)
```